packages/xod-project/src/pin.js

```diff
@@ -272,6 +273,31 @@ export const normalizeEmptyPinLabels = def(
   )
 );
 
+/**
+ * Normalizes pin label to C++ friendly label:
+ * - Space (" ") -> "_"
+ * - Minus ("-") -> "_"
+ * - Custom unicode ("δ") -> Unicode sequence ("U03B4")
+ * - Custom rare unicode ("🙀") -> "U1F640"
+ */
+export const cppEscape = def(
+  'cppEscape :: PinLabel -> PinLabel',
+  R.compose(unicodeCharsToUnicodeSequence, R.replace(/(\s|-)+/g, '_'), R.trim)
+);
+
+/**
+ * Gives pins C++ friendly labels:
+ * - Normalized empty pin labels
+ * - Normalized non-ASCII and forbidden chars
+ */
+export const normalizePinLabelsForCpp = def(
+  'normalizePinLabelsForCpp :: [Pin] -> [Pin]',
```

Comment:
If everything other is right, this may be safely made a private detail of `xod-arduino`